gf2_192/src/gf2_192poly.rs

```diff
@@ -0,0 +1,242 @@
+            self.coefficients[i].to_i8_slice(&mut res, i * 24).unwrap();
+        }
+        res
+    }
+
+    /// Adds r*p to `self`. Assumes:
+    ///  - p is monic
+    ///  - self.coefficients.len() > p.degree
+    ///  - p.degree == self.degree + 1 or (self == 0 and p == 1)
+    fn add_monic_times_constant(&mut self, p: Gf2_192Poly, r: Gf2_192) {
+        let mut _t = Gf2_192::new();
+        for i in 0..p.degree {
+            _t = p.coefficients[i] * r;
+            self.coefficients[i] = self.coefficients[i] + _t;
+        }
+        self.degree = p.degree;
+        self.coefficients[self.degree] = r;
+    }
+
+    /// Multiply `self` by `x + r`. Assumes that `self` is monic (i.e.
+    /// `self.coefficients[self.degree] == 1`)
+    fn multiply_by_linear_binomial(&mut self, r: i8) {
+        self.degree += 1;
+        self.coefficients[self.degree] = Gf2_192::from(1);
+        for i in (1..self.degree).rev() {
+            self.coefficients[i] = Gf2_192::mul_by_i8(self.coefficients[i], r);
+            self.coefficients[i] = self.coefficients[i] + self.coefficients[i - 1];
+        }
+        self.coefficients[0] = Gf2_192::mul_by_i8(self.coefficients[0], r);
+    }
+
+    /// Constructs a constant polynomial (degree 0) which takes value of `constant_term`.
+    /// `max_degree` specifies the maximum degree of the polynomial (to allocate space).
+    fn make_constant(max_degree: usize, constant_term: i32) -> Gf2_192Poly {
+        let mut coefficients: Vec<_> = std::iter::repeat_with(Gf2_192::new)
+            .take(max_degree + 1)
+            .collect();
+        coefficients[0] = Gf2_192::from(constant_term);
+        let degree = 0;
+        Gf2_192Poly {
+            degree,
+            coefficients,
+        }
+    }
+}
+
+impl<'a> TryFrom<CoefficientsByteRepr<'a>> for Gf2_192Poly {
+    type Error = Gf2_192Error;
+
+    fn try_from(c: CoefficientsByteRepr<'a>) -> Result<Self, Self::Error> {
+        let degree = c.more_coeffs.len() / 24;
+        let mut coefficients = Vec::with_capacity(degree + 1);
+        coefficients.push(Gf2_192::from(c.coeff0));
+
+        for i in 1..=degree {
+            coefficients.push(Gf2_192::try_from(&c.more_coeffs[(i - 1) * 24..])?);
+        }
+        Ok(Gf2_192Poly {
+            degree,
+            coefficients,
+        })
+    }
+}
+
+/// The following tests closely match those in `ScoreXFoundation/sigmastate-interpreter`.
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rand::{thread_rng, Rng};
+
+    #[test]
+    fn test_interpolation() {
+        // Try with arrays of length 0
+
+        // Constant polynomial with value 0
+        let mut p = Gf2_192Poly::interpolate(&[], &[], Gf2_192::new()).unwrap();
+        assert!(p.evaluate(0).is_zero());
+        assert!(p.evaluate(5).is_zero());
+
+        // Constant polynomial with value 17
+        let val_17 = Gf2_192::from(17);
+        p = Gf2_192Poly::interpolate(&[], &[], val_17).unwrap();
```

Comment:
Note that `sigmastate-interpreter` doesn't actually use empty arrays for these tests: https://github.com/ScorexFoundation/sigmastate-interpreter/blob/262a14d17ae3090fa97d1de1effd0b338562e4db/sigmastate/src/test/java/gf2t/GF2_192Test.java#L595-L599. Reyzin has a comment '0 is in the points array and valueAt0!=null, behavior is undefined', but these tests are counter-examples : ) 

Also the test on line 599 there is actually wrong! `p = GF2_192_Poly.interpolate(new byte[0], new GF2_192[0], val17);`. We can't have a polynomial taking values of 0 and 17 at zero! Strangely I assume the test is passing there, but it failed on my implementation.

Reply:
Good catch! I see that both implementation and tests are written by Reyzin. :) Maybe @kushti you can shed some light here?

Reply:
`new byte[0]` means empty array in Java.

thus `GF2_192_Poly.interpolate(new byte[0], new GF2_192[0], new GF2_192(0))` means that the only point among arguments which is defined is 0, and value is 0 

similarly, `GF2_192_Poly.interpolate(new byte[0], new GF2_192[0], val17)` means that f(0) = 17 and that is the only constraint 

so tests are correct

Reply:
> so tests are correct

Thank you for the explanation and for reviewing the PR!